pkg/identity/cache/allocation_test.go

```
@@ -325,19 +363,177 @@ func testAllocator(t *testing.T) {
+			allocationMode: option.IdentityAllocationModeDoubleWriteReadCRD,
+		},
+	}
+
+	for _, tc := range testCases {
+		testName := fmt.Sprintf("%s_%s", testNamePrefix, tc.name)
+		t.Run(testName, func(t *testing.T) {
+			option.Config.IdentityAllocationMode = tc.allocationMode
+			defer func() { option.Config.IdentityAllocationMode = option.IdentityAllocationModeKVstore }()
+
+			lbls1 := labels.NewLabelsFromSortedList("blah=%%//!!;id=foo;user=anna")
+
+			ctx := context.Background()
+			_, kubeClient := k8sClient.NewFakeClientset()
+
+			owner := newDummyOwner()
+			identity.InitWellKnownIdentities(fakeConfig, cmtypes.ClusterInfo{Name: "default", ID: 5})
+			mgr := NewCachingIdentityAllocator(owner, AllocatorConfig{EnableOperatorManageCIDs: true, maxAllocAttempts: 2})
+			<-mgr.InitIdentityAllocator(kubeClient)
+			defer mgr.Close()
+			defer mgr.IdentityAllocator.DeleteAllKeys()
+
+			// Verify that allocating an identity that doesn't exist will return an error.
+			id1a, isNew, err := mgr.AllocateIdentity(ctx, lbls1, false, identity.InvalidIdentity)
+			require.Nil(t, id1a)
+			require.Error(t, err)
+			require.Equal(t, false, isNew)
+
+			id := testCreateCIDObj("1000", lbls1)
+
+			var err2 error
+			switch option.Config.IdentityAllocationMode {
+			case option.IdentityAllocationModeKVstore:
+				err = addIDKVStore(ctx, id.Name, lbls1)
+			case option.IdentityAllocationModeCRD:
+				_, err = kubeClient.CiliumV2().CiliumIdentities().Create(ctx, id, metav1.CreateOptions{})
+			case option.IdentityAllocationModeDoubleWriteReadKVstore, option.IdentityAllocationModeDoubleWriteReadCRD:
+				err2 = addIDKVStore(ctx, id.Name, lbls1)
+				_, err = kubeClient.CiliumV2().CiliumIdentities().Create(ctx, id, metav1.CreateOptions{})
+			}
+			require.NoError(t, err)
+			require.NoError(t, err2)
+
+			// Verify that the created CID is allocated, as an existing CID in the store.
+			var id2 *identity.Identity
+			testutils.WaitUntil(func() bool {
```

`WaitUntil()` error are not handled

---

Done.

---

i think there are a few more occurrences

---

Done.